### ISSUE TITLE
'GasLimit' field cannot be greater than a block's gas limit!

### ISSUE THREAD

**Issue by dispensable1988** [filecoin-project/lotus#8997]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [X] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
Daemon:  1.16.0+mainnet+git.8a7806a.dirty+api1.5.0
Local: lotus version 1.16.0+mainnet+git.8a7806a.dirty
```

### Describe the Bug

We encountered a problem that caused the submission of the SubmitWindowedPoSt to fail which inclued  4 partitions,,
in the save deadline, Another message was submitted successfully which included 1 partition , the cid is `bafy2bzaced2dxsssaufsmr6io5pwinmcqmvljfanwgjpi7oo2r7aczzu2ueok` but the gaslimit is large too:
2,176,877,689

### Logging Information

```text
sign message err:mpool push: failed to push message: message not valid for block inclusion: 'GasLimit' field cannot be greater than a block's gas limit!
```

### Repo Steps

1. Run '...'
2. Do '...'
3. See error '...'
...

**Comment by dispensable1988** [comment]
lotus-miner  sectors  compact-partitions --deadline 17 --partitions 0 --partitions 1 --partitions 2 --partitions 3  --really-do-it
compacting 4 paritions
ERROR: mpool push: mpool push: failed to push message: message not valid for block inclusion: 'GasLimit' field cannot be greater than a block's gas limit

**Comment by jennijuju** [comment]
the original issue should be resolved by v1.16.1

**Comment by jennijuju** [comment]
> lotus-miner sectors compact-partitions --deadline 17 --partitions 0 --partitions 1 --partitions 2 --partitions 3 --really-do-it
> compacting 4 paritions
> ERROR: mpool push: mpool push: failed to push message: message not valid for block inclusion: 'GasLimit' field cannot be greater than a block's gas limit

this can be a future enhancement - please create a separate ticket.  for now you can compact 1 deadline each time.

**Comment by rjan90** [comment]
Closing this issue since it has been resolved by Lotus v1.16.1
